cmd/gpu_plugin/gpu_plugin.go

```diff
@@ -53,9 +54,88 @@ const (
 )
 
 type cliOptions struct {
-	sharedDevNum       int
-	enableMonitoring   bool
-	resourceManagement bool
+	preferredAllocationPolicy string
+	sharedDevNum              int
+	enableMonitoring          bool
+	resourceManagement        bool
+}
+
+type preferredAllocationPolicyFunc func(*pluginapi.ContainerPreferredAllocationRequest) []string
+
+// nonePolicy is used for allocating GPU devices randomly.
+func nonePolicy(req *pluginapi.ContainerPreferredAllocationRequest) []string {
+	klog.V(2).Info("Select nonePolicy for GPU device allocation")
+
+	deviceIds := req.AvailableDeviceIDs[0:req.AllocationSize]
+
+	klog.V(2).Infof("Allocate deviceIds: %q", deviceIds)
+	return deviceIds
+}
+
+// balancingPolicy is used for allocating GPU devices in balance.
+func balancingPolicy(req *pluginapi.ContainerPreferredAllocationRequest) []string {
```

Comment:
nit: there is a bit of inconsistency between user-facing naming (none, balanced, packed) and internal functions naming (none, balancing, packing). I'd suggest to rename balancingPolicy -> balancedPolicy and packingPolicy -> packedPolicty.